packages/application-config/src/transformers.ts

```diff
@@ -19,16 +22,49 @@ const computeUriPath = (uriPath: string, entryPointUriPath: string) => {
   return `${entryPointUriPath}/${uriPath}`;
 };
 
+type TResourceAccessPermissionName<PermissionName extends string = ''> =
+  | `view${Capitalize<PermissionName>}`
+  | `manage${Capitalize<PermissionName>}`;
+
+type TImplicitCustomApplicationResourceAccesses<
+  PermissionName extends string = ''
+> = Record<TResourceAccessPermissionName<PermissionName>, string[]>;
+
 function transformCustomApplicationConfigToData(
   appConfig: JSONSchemaForCustomApplicationConfigurationFiles
 ): CustomApplicationData {
   validateEntryPointUriPath(appConfig);
   validateSubmenuLinks(appConfig);
```

Comment:
We should validate a couple more things:
* The additional permission names should be validated the same as the entry point.
* There should be no duplicated permission names (similar to the sub-links).